### ISSUE TITLE
Mask error areas while unwrapping

### ISSUE THREAD

**Issue by comma-never-coma** [isce-framework/isce2#569]
There are two methods for phase unwrapping in `stackSentinel.py`, ICU and SNAPHU.
I find the “Integrated Correlation and Unwrapping” (ICU) method can mask large areas. However, I can not find any parameter in SNAPHU to mask unreliable areas though the program will generate a conncomp file which can be used to mask areas after unwrapping. I wonder if areas with low coherence could be masked before phase unwrapping.
Here are two images after unwrapping, the water body is still remained in unwrapped phase and brings many noise to the phase map.
![20200911_20200923](https://user-images.githubusercontent.com/67263104/186129392-40d14b89-4e17-4c2d-9856-e946105e5123.jpg)
![20200911_20200923 unw](https://user-images.githubusercontent.com/67263104/186129134-4bbcf270-265a-48b6-94a2-50fba5d7fe4d.jpg)

I tried to run `wbd.py` & `waterMask.py` and failed, the scripts seemed to be disabled. Thus, how can I mask low coherent areas or water bodies before SNAPHU to reduce phase unwrapping errors. 
Any references or advice is appreciated!

**Comment by jerry7nol** [comment]
do you resolve it

**Comment by jerry7nol** [comment]
i meet same question.and people of using ISCE to do DINSAR seem few . do you solved it?

**Comment by comma-never-coma** [comment]
> i meet same question.and people of using ISCE to do DINSAR seem few . do you solved it?

After version 2.0, snaphu provides a masking option. The following content is excerpted from 'README_releasenotes.txt' in snaphu-v2.0.6:
  

> The user can indicate masking by setting the magnitudes of the complex values of pixels to be masked to zero in the input file or by specifying a separate binary mask via the new -M option or the new BYTEMASKFILE keyword.  See the comments in the template configuration file for the BYTEMASKFILE keyword for a full description of the file format.  A fixed number of pixels at each edge of the input file can also be masked via the EDGEMASKTOP, EDGEMASKBOT, EDGEMASKLEFT, and EDGEMASKRIGHT keywords.

However, ISCE2 still defaults to using snaphu-v1.4, so it is necessary to compile snaphu v2.0 on your own and modify the source code in ISCE regarding the unwrapping parameters to read the mask file required by the -M option in snaphu-v2.0.

**Comment by jerry7nol** [comment]
> > i meet same question.and people of using ISCE to do DINSAR seem few . do you solved it?
> 
> After version 2.0, snaphu provides a masking option. The following content is excerpted from 'README_releasenotes.txt' in snaphu-v2.0.6:
> 
> > The user can indicate masking by setting the magnitudes of the complex values of pixels to be masked to zero in the input file or by specifying a separate binary mask via the new -M option or the new BYTEMASKFILE keyword.  See the comments in the template configuration file for the BYTEMASKFILE keyword for a full description of the file format.  A fixed number of pixels at each edge of the input file can also be masked via the EDGEMASKTOP, EDGEMASKBOT, EDGEMASKLEFT, and EDGEMASKRIGHT keywords.
> 
> However, ISCE2 still defaults to using snaphu-v1.4, so it is necessary to compile snaphu v2.0 on your own and modify the source code in ISCE regarding the unwrapping parameters to read the mask file required by the -M option in snaphu-v2.0.

thank you comma

**Comment by yuankailiu** [comment]
Hi @comma-never-coma, just a question from me. I am also interested in applying mask before SNAPHU unwrapping. Based on that readme file, 
> The user can indicate masking by setting the magnitudes of the complex values of pixels to be masked to zero in the input file
> Areas with zero magnitude in the input data are treated as masked areas as well.

Is this also a new feature in ver-2.0.5? Or this is also true in ver-1.4 in the current ISCE2?
I.e., can we simply set the water region to have zero magnitude and SNAPHU 1.4 in current ISCE2 will ignore those areas?

**Comment by comma-never-coma** [comment]
Yes you are right @yuankailiu . We can indeed mask areas by setting the magnitude of complex values of pixels in the water regions to zero in the current ISCE2 with SNAPHU 1.4. This feature can be found within the 'snaphu_cost.c' file of SNAPHU as shown below.

```
      /* see if we have a masked pixel */
      if(mag[row][col]==0 || mag[row+1][col]==0){

	/* masked pixel */
	rowcost[row][col].laycost=0;
	rowcost[row][col].offset=LARGESHORT/2;
	rowcost[row][col].dzmax=LARGESHORT;
	rowcost[row][col].sigsq=LARGESHORT;
```

Additionally, it is important to note that even if the magnitude of pixels is set to zero for certain regions, the phase will still be present in those areas in the generated unwrapped interferogram. However, I believe this approach is beneficial as it helps to reduce the impact of noisy areas on the overall accuracy of phase unwrapping. After completing the phase unwrapping process, we can mask phase values in these zero magnitude areas with NAN.

**Comment by Sci123-lab** [comment]
@comma-never-coma,
Thanks for your comments. I am wondering if its possible to have water mask applied before going to unwrapping using StripmapApp.  I am using ALOS palsar images and have developed interferogram but currently,  I wish to be able to have water mask applied prior to unwrapping and ionospheric correction.

I downloaded .wbd watermask using wbd.py  based on coordinates of my area, however, both dem and watermask are seperate files. I tried wbdStitcher but it isn't working for me. > I will appreciate your guidance on this matter, thankss :)
